lib/bson/decimal128.rb

```diff
@@ -107,11 +107,18 @@ def ==(other)
     #
     # @since 4.2.0
     def initialize(initial, digits = nil)
-      if initial.is_a?(String)
-        set_bits(*Builder::FromString.new(initial).bits)
-      else
-        big_decimal = BigDecimal.new(*[initial, digits].compact)
-        set_bits(*Builder::FromBigDecimal.new(big_decimal).bits)
+      case initial
+        when String
+          set_bits(*Builder::FromString.new(initial).bits)
+        when Integer
+          set_bits(*Builder::FromInteger.new(initial).bits)
+        when Float
+          set_bits(*Builder::FromFloat.new(initial).bits)
```

Comment:
but I extract the number of sig digits here, is it really necessary? https://github.com/mongodb/bson-ruby/pull/61/files/2ef994a4d637fb1187d598db63a5277139b0e5ef..b3a4cdd1a82c43eb431558084647b9a27e535453#diff-87a1887ce145bf861a2df29a0ac56b82R286